### PR TITLE
Rebalance jobs

### PR DESCRIPTION
**Pull request by Armavica**
**What is this PR about?**
This PR rebalances the GitHub test workers as discussed in #6170.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [ ] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- ...

### CODE REVIEW

File: scripts/check_all_tests_are_covered.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def from_yaml():
     # First collect the matrix definitions from testing workflows
     matrices = {}
     for wf in ["tests.yml"]:
-        wfname = wf.strip(".yml")
+        wfname = wf.removesuffix(".yml")
```
Comment (ricardoV94) [inline scripts/check_all_tests_are_covered.py]: This was only introduced in `3.9` https://docs.python.org/3/library/stdtypes.html#str.removesuffix

Comment (Armavica) [reply scripts/check_all_tests_are_covered.py]: Good catch!

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Some conflict. I added a test file in a PR that was just merged, so probably straightforward

**Comment by ricardoV94** [comment]
Thanks @Armavica! Should we consider not running that very slow macos job at all?

**Comment by Armavica** [comment]
Currently it is not taking that much longer than the longest of the other runners, so it wouldn't make a big difference (just a couple of minutes).

**Pull request merged**